apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx

```diff
@@ -274,6 +293,33 @@ export const StepBasedFlow: FunctionComponent<AuthenticationFlowPropsInterface>
         onAuthenticationSequenceChange(true, authenticationSteps);
     }, [ authenticationSteps ]);
 
+    /**
+     * This identifies what should be the subject identifier for the current authentication steps and set it as
+     * subject id. This resolves the issue of subject identifier always been step 01.
+     */
+    useEffect(() => {
+        let hasFoundTheSubjectIdentifiableStep: boolean = false;
```

Comment:
We can extract this to a function and reuse IMO. WDYT?

Reply:
On a second thought, I feel like extracting this piece of code into a function is not necessary.